### ISSUE TITLE
Doxygen Refresh

### ISSUE THREAD

**Issue by jackoalan** [libAthena/Athena#35]
Athena has changed considerably since the doxygen comments were last updated.
Several APIs (notably the DNA and YAML classes) have no docs at all.

This issue tracks the refresh/update of these docs starting from 2.4.0 release.

**Comment by Antidote** [comment]
We should start by documenting the DNA structs, since those are completely devoid of them. Then we'll use that as our new document standard.

**Comment by jackoalan** [comment]
Agreed, we can also fill up the org's Jekyll site with usage examples

**Comment by Antidote** [comment]
Yeah, Jekyll should provide usage examples, while the doxygen should simply document use cases in general.
